src/Rule.js

```diff
@@ -75,29 +79,36 @@ class Rule {
-     * types:
+     * Return the type of intermediate representation that this rule can process. Rules can
+     * be any type as long as there is a parser that produces that type. By convention,
+     * there are a few types that are already defined:
      *
-     * - A resource rule. This checks a translated resource with a source string
-     *   and a translation to a given locale. eg. a rule that checks that
+     * - resource - This checks a translated Resource instance with a source string
+     *   and a translation string for a given locale. For example, a rule that checks that
      *   substitution parameters that exist in the source string also are
-     *   given in the target string.
-     * - A line rule. This rule checks single lines of a file. eg. a rule to
+     *   given in the target string. Typically, resource files like po, properties, or xliff
+     *   are parsed into an array of Resource instances as its intermediate representations.
+     * - line - This rule checks single lines of a file. eg. a rule to
      *   check the parameters to a function call.
-     * - A multiline rule. This rule checks multiple lines at once to find
-     *   problems that may span multiple lines. For example, a rule to enforce
-     *   a policy that all translatable strings in a source file have an appropriate
-     *   translator's comment on them.
-     * - A multifile rule. This rule checks problems across multiple files. eg.
-     *   a rule to check that ids for translatable strings are unique across all
-     *   files.
-     *
-     * @returns {String} a string with either "resource", "line", "multiline", or
-     * "multifile".
+     * - string - This rule checks the entire file as a single string. Often, this type
+     *   of representation is used with source code files that are checked with regular
+     *   expressions, which often mean declarative rules.
+     * - {other} - You can choose to return any other string here that uniquely identifies the
+     *   representation that a parser produces.
+     *
+     * Typically, a full parser for a programming language will return something like
+     * an abstract syntax tree as an intermediate format. For example, the acorn parser
+     * for javascript returns an abstract syntax tree in JSTree format. The parser may
+     * choose to return the string "ast-jstree" as its identifier, as long as there are
+     * rules that are looking for that same string. The parser can return any string it
+     * likes just as long as there are rules that know how to check it.
+     *
+     * @returns {String} a string that names the type of intermediate representation
+     * that this rule knows how to check
      */
     getRuleType() {
-        // default rule type. If your rule is different, override this method.
-        return "line";
+        // default representation type. If your rule is different, override this method.
+        return "string";
```

Comment:
There should not be a default expected representation type, because there is no default rule implementation.